### DISCUSSION TITLE
Parameters passed to URL::defaults(), are not using "bindingFieldFor" when creating routes.

### DISCUSSION THREAD

**Discussion by demobiel** [laravel/framework#49017] [Q&A]
Hi,

we are trying to assign models to the url defaults, the same way you can pass a model as parameter to the route function. 
However we discovered this is not working.

What we try to achieve
//$event is an instance of App\Models\Event::class.
        URL::defaults([
            'event' => $event,
        ]);

However we found out that the url defaults are just assigned as is, in this case we get the JSON string of the event model.

Example:
Route::get('/events/{event:slug}/orders/{order:uuid}',[OrderController::class,'show'])->name('event.order.show');

route('event.order.show',['order' => $order ]);

->result 'events/{"id":1,"name":"eventname", "slug":"event-slug"}/orders/8f07a5bf-d517-4e2e-8f8e-c5d6eac18d46'

What we found as possible fix (but not sure if this causes any other trouble), if we adapt line Illuminating\Routing\UrlGenerator.php:494
        $parameters = collect(Arr::wrap($parameters))->map(function ($value, $key) use ($route) {
    to
        $parameters = collect(array_merge($this->getDefaultParameters(),Arr::wrap($parameters)))->map(function ($value, $key) use ($route) {

we get as expected result
'events/event-slug/orders/8f07a5bf-d517-4e2e-8f8e-c5d6eac18d46'

is this a missing feature / bug on Laravel's side, or are we abusing the URL::defaults feature? 

Kind regards,

Andries

**Comment by demobiel** [comment]
sorry... the formatting got removed... here it is with formatting

Hi,

we are trying to assign models to the url defaults, the same way you can pass a model as parameter to the route function.
However we discovered this is not working.

What we try to achieve
```
//$event is an instance of App\Models\Event::class.
URL::defaults([
'event' => $event,
]);
```
However we found out that the url defaults are just assigned as is, in this case we get the JSON string of the event model.

Example:
```
Route::get('/events/{event:slug}/orders/{order:uuid}',[OrderController::class,'show'])->name('event.order.show');

route('event.order.show',['order' => $order ]);

->result 'events/{"id":1,"name":"eventname", "slug":"event-slug"}/orders/8f07a5bf-d517-4e2e-8f8e-c5d6eac18d46'
```

What we found as possible fix (but not sure if this causes any other trouble), if we adapt line Illuminating\Routing\UrlGenerator.php:494
```
$parameters = collect(Arr::wrap($parameters))->map(function ($value, $key) use ($route) {
to
$parameters = collect(array_merge($this->getDefaultParameters(),Arr::wrap($parameters)))->map(function ($value, $key) use ($route) {
```
we get as expected result
`'events/event-slug/orders/8f07a5bf-d517-4e2e-8f8e-c5d6eac18d46'` is this a missing feature / bug on Laravel's side, or are we abusing the URL::defaults feature for something they are not intended for?

Kind regards,

Andries

**Reply by shaedrich** [reply]
Can't you edit the question instead of reposting it as an answer?

**Reply by demobiel** [reply]
I am not allowed to edit the content my question... only the title, otherwise I would have done it :)

**Reply by shaedrich** [reply]
Oh, sorry, too bad. Would make sense if one could edit their own question 😃

**Comment by shaedrich** [comment]
I'm not sure if the quite generic `URL` class is the proper place for Laravel-specific model logic. Might be better to have a `Route::defaults()` instead. That would be the same as `request()` vs `route()`.

**Reply by demobiel** [reply]
I understand what you are saying, but at the moment the docs specifies the use of the URL class for defaults.
https://laravel.com/docs/10.x/routing#generating-urls-to-named-routes

**Reply by shaedrich** [reply]
That's right. However, I wouldn't consider that a bug per se. But that might be up to people like Taylor to decide on that.

**Reply by demobiel** [reply]
I am not too familiar (unfortunately) with the inners of the laravel framework to decide where this should be implemented, do you think I should rise an issue for this, or just keep it here in the discussion area?

**Reply by shaedrich** [reply]
In my opinion is that it's usually not worth raising an issue in this repository since chances are high, it will be rejected anyway, and you'll be lucky if the explanation is even remotely conclusive.
